src/pineko/theory.py

```diff
@@ -380,18 +381,29 @@ def fk(self, name, grid_path, tcard, pdf):
         max_al = 0
         # check for sv
         if not np.isclose(xir, 1.0):
-            is_ren_as, is_ren_al = check.contains_ren(grid, max_as, max_al)
-            if not (is_ren_as and is_ren_al):
-                raise ValueError(
-                    "Renormalization scale variations are not available for this grid"
-                )
+            checkres, max_as_effective = check.contains_sv(
+                grid, max_as, max_al, check.Scale.REN
+            )
+            if max_as == max_as_effective:
+                if checkres is check.AvailableAtMax.SCVAR:
+                    raise ValueError("Central order is not available but sv order is.")
+            if max_as < max_as_effective and checkres is not check.AvailableAtMax.BOTH:
+                raise ValueError("No available central order or sv order.")
         if sv_method is None:
             if not np.isclose(xif, 1.0):
-                is_fact_as, is_fact_al = check.contains_fact(grid, max_as, max_al)
-                if not (is_fact_as and is_fact_al):
-                    raise ValueError(
-                        "Factorization scale variations are not available for this grid"
-                    )
+                checkres, max_as_effective = check.contains_sv(
+                    grid, max_as, max_al, check.Scale.FACT
+                )
+                if max_as == max_as_effective:
+                    if checkres is check.AvailableAtMax.SCVAR:
+                        raise ValueError(
+                            "Central order is not available but sv order is."
+                        )
+                if (
+                    max_as < max_as_effective
+                    and checkres is not check.AvailableAtMax.BOTH
+                ):
+                    raise ValueError("No available central order or sv order.")
```

Comment:
If I'm not wrong, this is exactly the same as L384-391, with the only difference of `check.Scale.REN` being traded for `check.Scale.FACT`.

You know how to do it: split a function and call twice ;)